pkg/kudoctl/cmd/init/manager.go

```diff
@@ -0,0 +1,218 @@
+package init
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/version"
+
+	"k8s.io/api/apps/v1beta2"
+	v1 "k8s.io/api/core/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	clientv1beta2 "k8s.io/client-go/kubernetes/typed/apps/v1beta2"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/yaml"
+)
+
+//Defines the deployment of the KUDO manager and it's service definition.
+
+const (
+	group              = "kudo.dev"
+	crdVersion         = "v1alpha1"
+	defaultns          = "kudo-system"
+	defaultGracePeriod = 10
+)
+
+// Options is the configurable options to init
+type Options struct {
+	// Version is the version of the manager `0.5.0` for example (must NOT include the `v` in `v0.5.0`)
+	Version string
+	// Namespace to init into (default is kudo-system)
+	Namespace string
+	// TerminationGracePeriodSeconds defines the termination grace period for a pod
+	TerminationGracePeriodSeconds int64
+	// Image defines the image to be used
+	Image string
+}
+
+// NewOptions provides an option struct with defaults
+func NewOptions(v string) Options {
+
+	if v == "" {
+		v = version.Get().GitVersion
+	}
+
+	return Options{
+		Version:                       v,
+		Namespace:                     defaultns,
+		TerminationGracePeriodSeconds: defaultGracePeriod,
+		Image:                         fmt.Sprintf("kudobuilder/controller:v%v", v),
+	}
+}
+
+// Install uses Kubernetes client to install KUDO manager.
+func Install(client kubernetes.Interface, extClient *apiextensionsclient.Clientset, opts Options) error {
+	if err := installPrereqs(client, opts); err != nil {
+		return err
+	}
+	if err := installCrds(extClient); err != nil {
+		return err
+	}
+	if err := installManager(client, opts); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Install uses Kubernetes client to install KUDO manager.
+func installManager(client kubernetes.Interface, opts Options) error {
+
+	if err := installStatefulSet(client.AppsV1beta2(), opts); err != nil {
+		return err
+	}
+
+	if err := installService(client.CoreV1(), opts); err != nil {
+		return err
+	}
+	return nil
+}
+
+func installStatefulSet(client clientv1beta2.StatefulSetsGetter, opts Options) error {
+	ss := generateDeployment(opts)
+	_, err := client.StatefulSets(opts.Namespace).Create(ss)
+	return err
+}
+
+func installService(client corev1.ServicesGetter, opts Options) error {
+	s := generateService(opts)
+	_, err := client.Services(opts.Namespace).Create(s)
+	return err
+}
+
+// DeploymentManifests provides a slice of strings for the deployment and service manifest
+func DeploymentManifests(opts Options) ([]string, error) {
```

Comment:
could we find better name? I have hard time figuring out what deployment, should this rather be `ManagerManifests` or ... ?